pageserver/src/page_service.rs

```diff
@@ -934,24 +991,58 @@ impl PageServerHandler {
                 };
                 let span = mkspan!(shard.tenant_shard_id.shard_slug());
 
-                let timer = record_op_start_and_throttle(
-                    &shard,
-                    metrics::SmgrQueryType::GetPageAtLsn,
-                    received_at,
-                )
-                .await?;
+                get_page_context.perf_span_record(
+                    "shard",
+                    tracing::field::display(shard.get_shard_identity().shard_slug()),
+                );
```

Comment:
Not obvious to me where that `shard` field is defined. Remember that
1. `record()` for an undefined field is a silent no-op and
2. `record()` is a silent no-op if the field is defined in a parent span, but you call record() on a child span.

Reply:
Indeed, it's defined on the parent span [here](https://github.com/neondatabase/neon/blob/ca789c5e840559c2dc5af803337b0f69c092dd50/pageserver/src/page_service.rs#L941).

I had the same expectations as you (1 and 2), but 2 doesn't seem to be true. The field does show up on the parent.
I kept it there because it's nicer to see the shard field on the top level span.

Reply:
Nope. Will update.

Reply:
https://github.com/neondatabase/neon/pull/11140/commits/4311097d73df06e4ee0962b5f94ab3097e833386